### PR TITLE
List some new tokens

### LINKED ISSUE #78: Please list PLUM token
YAML:

addr: '0x41104c8d51a80d9e2969dc23256a352de4b0ec33'
decimals: 18
description: Payment Layer for Unattended Merchandising
name: PLUM
symbol: PLUM

Revised 3/8/18

0x41104c8d51a80d9e2969dc23256a352de4b0ec33

http://42plums.com

https://etherscan.io/address/0x41104c8d51a80d9e2969dc23256a352de4b0ec33

www.reddit.com/r/plumcoin

### LINKED ISSUE #177: HODLER.TECH token (HDL)
Hello.
For 7 days we will start our ICO with the new hardware wallet. Could you add our token on Forkdelta?
Information about the token:
1. The contract address: 0x95C4be8534d69C248C0623c4C9a7A2a001c17337  
2. Official token website: https://hodler.tech
3.The ultimate Hardware Wallet with the most advanced level of security, designed for easy payments with over 350 cryptocurrency.
4. https://hodler.tech/f/ICOdetails.html#tkndetails    and https://ethplorer.io/address/0x95c4be8534d69c248c0623c4c9a7a2a001c17337
5. We will start the bitcointalk thread for 30 Feb 2018.

Many thanks in advance and best regards.
Tobiasz Horodko CEO of the HODLER.TECH LTD company

### LINKED ISSUE #341: Request to list Rezox Token on ForkDelta
Contract address: 0xc7a4eb66f982cde71d872398e4d1fc5fc586cb9e

Issuer's official website: www.rezox.network

Official contract address confirmation: Can be found at main page here: www.rezox.network
The address is 0xc7a4eb66f982cde71d872398e4d1fc5fc586cb9e

A link to a third-party review or discussion of the token: https://bitcointalk.org/index.php?topic=2912627

Description: Rezox Token, is a semi-mineable Token thats based on the Ethereum network.
Our goal is to promote low-end computers and make them be able to mine and get our Tokens and take part in the Blockchain revulotion.

More information:
Decimals: 8
links:
- Telegram: https://t.me/rezoxtoken
Name: Rezox
Symbol: REZ

### LINKED ISSUE #342: Request to list GUESS Token on ForkDelta
Contract address: 0xbdcfbf5c4d91abc0bc9709c7286d00063c0e6f22

name : GUESS
symbol : GUESS

Issuer's official website: https://peerguess.com/

Official contract address confirmation:
https://peerguess.com/

A link to a third-party review or discussion of the token:
 https://bitcointalk.org/index.php?topic=2199333.0 

Description:

Peerguess is a free, one-stop-shop cryptocurrency
price ticker application where the users will also guess
future prices to earn gems, learn about community
tendencies and have access to the sophisticated data
to improve your intuition. With collaboration at the
heart of the system, peerguess will provide “market
sentiment” data to both its users and business
partners.

### LINKED ISSUE #345: RevizorCoin (RR) listing request
Contract address: 0x5fB6a842Ea587fa8d0486589742Ce68Ef24834A4

Issuer's official website: [https://revizor-coin.io/](url)

Official contract address confirmation: [https://revizor-coin.io/#about](url)

A link to a third-party review or discussion of the token: [https://bitcointalk.org/index.php?topic=2645567.0](url)

Description: 
Revizor is an innovative project in the field of regulation of the crypto-currency market, the main task of which is to create software tools that would filter a wide range of assets by absorbing unpromising tokens. 

_________________________________________________________________________
addr: 0x5fb6a842ea587fa8d0486589742ce68ef24834a4
decimals: 18
links:
- Email: rvr@revizor-coin.io
- Telegram: [https://t.me/revizorcoin](url)
- Twitter: [https://twitter.com/RevizorCoin_RR?lang=ru](url)
- Website: [revizor-coin.io](url)
- Whitepaper: [https://revizor-coin.io/WHITE_PAPER_ENG.pdf](url)
name: Revizor Coin
symbol: RR

### LINKED ISSUE #347: Request to add LaLa token on forkdelta
Contract address: 0xfd107b473ab90e8fbd89872144a3dc92c40fa8c9
Symbol: LALA
Decimals: 18

Issuer's official website: https://lalaworld.io

Official contract address confirmation: 
https://blog.lalaworld.io/add-lala-token-step-wise-guide/

A link to a third-party review or discussion of the token:
https://bitcointalk.org/index.php?topic=2268691.0

Description:
LALA World (“LALA”) is all about migrants and their unbanked families. Its an ECOSYSTEM enabled by the LALA Wallet as a PLATFORM, a single sign-on platform revolving around their issues and problems. Employment issues, digital ID’s, communities build up, government and NGO partnerships, health issues, and of course, a new financial ecosystem for the huge underbanked population by capitalizing the Blockchain revolution to bridge the gap between cash, digital world.

### LINKED ISSUE #355: ERC 20 Token Submission - CLL (CryptoLiveLeak)
Contract address: 0x3dc9a42fa7afe57be03c58fd7f4411b1e466c508

Issuer's official website: https://www.cryptoliveleak.com/

Official contract address confirmation:
https://www.cryptoliveleak.com/about-us
Scroll Down to the F.A.Q. section and look at the third Q/A.

A link to a third-party review or discussion of the token:
Please see blow videos about our CLL token - participants comments are on the videos.
https://youtu.be/6XuUdG7cqqs
https://youtu.be/Au8daAT0drE
https://youtu.be/t2t5WxOC8tE
Please let me know if you need anything else.

Description:
CryptoLiveLeak is a leading cryptocurrency news and media platform. We have integrated the CLL token with our platform to innovative ways the audience and media interact. The CLL Token has multiple benefits including exclusive discounts when participating in future ICOs from founder or through the CryptoLiveLeak platform. We have a large dedication to giving the CLL tokens to those who interact with our platform as show by our Airdrop and Giveaways. Token holders can also accumulate a LEAK SCORE to gain additional incentives when interacting with the CryptoLiveLeak platform. We aim to innovate and disrupt in a variety of industries and want our early supporters to gain with us by holding CLL token. This all occurs within a trustless environment where all participants have transparency. The platform will evolve as we progress with IBO efforts. The CryptoLiveLeak team will also look to expand and grow as the network matures. Large budget for development and marketing. The CryptoLiveLeak (CLL) ERC20 token will give holders an unique way of participating in the evolution of our vision of blockchain tech including transforming the payment and exchanges industries.

Address: 0x3dc9a42fa7afe57be03c58fd7f4411b1e466c508
Decimal: 18
Description: Token to interact with CryptoLiveLeak, Cryptocurrency News and Media Platform
Name: CryptoLiveLeak
Symbol: CLL
Links:
- Email: ContactUs@cryptoliveleak.com
- Telegram: https://t.me/CryptoLiveLeak
- Twitter: https://twitter.com/CryptoLiveLeak
- Website: www.cryptoliveleak.com
- Whitepaper: https://github.com/CryptoLiveLeak/CLL-Token/blob/master/CryptoLiveLeak%20WhitePaper.pptx
- Instagram: https://www.instagram.com/cryptoliveleak/
- Facebook: https://www.facebook.com/CryptoLiveLeak/
- Reddit: https://www.reddit.com/r/CryptoLiveLeak/

### LINKED ISSUE #361: Listing request - TOMO
Website: https://www.tomocoin.io
Whitepaper: https://drive.google.com/file/d/1xrtNX9y_ayyx7UKWYzIWOqley3jUUIWH/view
Description: We're building an efficient blockchain infrastructure for decentralized applications, token issuance and integration.

Contract address: https://etherscan.io/token/0x8b353021189375591723e7384262f45709a3c3dc
Symbol: TOMO
Full Name: Tomocoin
Decimals: 18

Link to confirm contract: https://twitter.com/TomoCoin_io/status/969457886798557184 (this is their official Twitter account, which is linked on their official website. You can also scroll all the way down to their official website and you'll find this tweet there too.)

Third-party comment(s) on the token:
https://bitcointalk.org/index.php?topic=2021785
https://cryptobriefing.com/tomocoin-ico-review-tmc-token-analysis

### LINKED ISSUE #362: Listing request - DADI
Website: https://dadi.cloud
Whitepaper: https://dadi.cloud/public/files/dadi.white-paper.pdf
Description: A new era of cloud computing services, powered by blockchain technology.

Contract address: https://etherscan.io/token/0xfb2f26f266fb2805a387230f2aa0a331b4d96fba
Symbol: DADI
Full Name: DADI
Decimals: 18

Link to confirm contract: https://steemit.com/dadi/@myblockchainhub/dadi-ico-pre-sale-and-public-sale-consolidated-info (go all the way to the bottom and you'll see they list the ethplorer address there which confirms this as the token address). Also, a DADI team member confirmed this address on Reddit: https://www.reddit.com/r/DADI/comments/7xywp5/right_on_cue_weve_started_token_distribution_for/dudt85y

Third-party comment(s) on the token:
https://bitcointalk.org/index.php?topic=2349363
https://icoranker.com/ico/dadi

### LINKED ISSUE #370: Listing request - Blocklancer
Contract address: 0x63e634330A20150DbB61B15648bC73855d6CCF07

Issuer's official website: www.blocklancer.net

Official contract address confirmation:

The address can be found here: https://blocklancer.net/tokensale/buy

A link to a third-party review or discussion of the token:

Bitcointalk: https://bitcointalk.org/index.php?topic=1974481.0
ICOBench: https://icobench.com/ico/blocklancer
ICOBazaar: https://icobazaar.com/v2/blocklancer/

Description:
Blocklancer is a Distributed Autonomous Job Marketplace (DAJ) operating on the Ethereum Blockchain. Blocklancer is our collaborative vision of an entirely self-regulatory platform for finding clients/projects and ensuring jobs are completed efficiently and fairly. Blocklancer is changing the way freelancing works for both the freelancer and the customer, concentrating on building a trustworthy platform all stakeholders can rely on. Blocklancer does this by minimalizing fees and introducing a decentralized tribunal system to guarantee every dispute case is settled fairly and that no one loses out — we can finally wave goodbye to fraud in freelancing.

### LINKED ISSUE #372: Listing request - WIZ
Website: https://crowdwiz.io
Whitepaper: https://crowdwiz.io/docs/whitepaper-en-1507561917.pdf
Description: The WizFund platform is a liberated ecosystem of investors, bringing flexibility and freedom through advanced technology. Every member has the right to vote and take the decision where and how the fund should invest. Token-holders are free to invest in newly-created funds or establish WizFunds of their own. A software wizard will guide you through the process. 

Contract address: https://etherscan.io/token/0x2f9b6779c37df5707249eeb3734bbfc94763fbe2
Symbol: WIZ
Full Name: CrowdWizToken
Decimals: 18

Link to confirm contract: https://medium.com/@Crowdwiz.io/how-to-claim-wiz-tokens-to-your-erc-20-compatible-wallet-2c20de5e0298

Third-party comment(s) on the token:
https://bitcointalk.org/index.php?topic=2175726
https://venturebeat.com/2017/09/13/blockchain-startup-crowdwiz-puts-a-decentralized-spin-on-fund-management

### LINKED ISSUE #373: Listing request - CND
Website: https://cindicator.com
Whitepaper: https://cindicator.com/Cindicator_WhitePaper_en.pdf
Description: Your personal Cryptocurrency Market measurement tool for spotting time-limited market opportunities.

Contract address: https://etherscan.io/token/0xd4c435f5b09f855c3317c8524cb1f586e42795fa
Symbol: CND
Full Name: Cindicator
Decimals: 18

Link to confirm contract: https://cindicator.com/add-metamask.pdf

Third-party comment(s) on the token:
https://bitcointalk.org/index.php?topic=2062875
https://crushcrypto.com/analysis-of-cindicator
https://bitcoinexchangeguide.com/cindicator

### LINKED ISSUE #375: Listing request - LGO
Website: https://legolas.exchange
Whitepaper: https://legolas.exchange/static//public/legolas-whitepaper-en.pdf?v=13
Description: Legolas Exchange creates a new decentralized blockchain protocol that guarantees full transparency and prevents front-running and market manipulation.

Contract address: https://etherscan.io/token/0x123ab195dd38b1b40510d467a6a359b201af056f
Symbol: LGO
Full Name: LGO Token
Decimals: 8

Link to confirm contract: https://twitter.com/LegolasExchange/status/969888161848586241 (their official Twitter account)

Third-party comment(s) on the token:
https://bitcointalk.org/index.php?topic=2383397
https://cointelegraph.com/news/big-investors-coming-a-new-decentralized-exchange-makes-plans-to-lure-them-all

### LINKED ISSUE #383: listing of Flash Token
Contract address:
https://ethplorer.io/address/0x25c8ca98fe9d59ceef5e0175abae40d4caa60af9
Issuer's official website:
https://flashtoken.io
Official contract address confirmation:

https://github.com/flashtoken
https://ethplorer.io/address/0x25c8ca98fe9d59ceef5e0175abae40d4caa60af9
A link to a third-party review or discussion of the token:

https://bitcointalk.org/index.php?topic=2574672.140
https://t.me/flashcommunityru

Description:

ticker is FTOKEN

### LINKED ISSUE #388: Meridian (MDN) - Contract Update
We have issued a new contract.

Old address: 0x550879F1778b575DDC8EB2095fBC9f4783b5f2F6

New Address: 0x1a61250a2bea646478184730d25648811bd429fe
Announcement of change: https://bitcointalk.org/index.php?topic=2157860.msg31653641#msg31653641
Website: https://mymdn.io/terms/

Many thanks

### LINKED ISSUE #392: Please list Stashpay (STP)
Contract address: 0xecd570bBf74761b960Fa04Cc10fe2c4e86FfDA36

Issuer's official website: https://stashpay.io/

Official contract address confirmation:  https://twitter.com/stashpay/status/951637472387121157

A link to a third-party review or discussion of the token: https://bitcointalk.to/index.php?topic=2114814

Description:
Stash is a next generation open source digital currency for the real world. A key advantage of Stash is that it solves numerous problems being encountered by other cryptocurrencies. Stash does this by bringing together many ground-breaking blockchain technologies while using the gold standard in cryptocurrency privacy.

In addition to comprising critical features for an efficiently functioning open decentralized autonomous network such as self-governance and self-funding, Stash also incorporates instantaneous transactions (StashSwift), increased network capacity, low transaction fees and state-of-the-art zero-knowledge cryptography.

### LINKED ISSUE #400: LISTING REQUEST - THUG COIN
1. Description:

THUG COIN 

Connecting thugs and your gang on the blockchain

Using the breakthrough technology of anonymous cryptocurrencies and the blockchain ledger. Real thugs themselves developed THUG Coin as the official cryptocurrency of the thug life around the world. THUG Coin acquires its uniqueness by providing thugs at every end of the globe with worldwide access to other thugs and gangs faster than the speed of light. Do not be fooled… This cryptocurrency is serious sh*t!

Quick facts:

- almost 50K members on Telegram
- over 5K hodlers on etherscan already

2. Contract address:

CONTRACT: 0xfe7B915A0bAA0E79f85c5553266513F7C1c03Ed0
DECIMALS: 18
TOKEN SYMBOL: THUG

3. Issuer's official website:

https://thugcoin.cash/

4. Official contract address confirmation:

Open our website (https://thugcoin.cash/) and you see the contract address icon on your screen. Moreover the contract address can be found in the FAQ section of the website.

5. A link to a third-party review or discussion of the token:

https://bitcointalk.org/index.php?topic=3077297

### PR DESCRIPTION
**Pull request by JonathonDunford**
List:
- PLUM
- HDL
- REZ
- GUESS
- RR
- LALA
- CLL
- TOMO
- DADI
- LANCER
- WIZ
- CND
- LGO
- FTOKEN
- STP
- THUG

Update:
- MDN

Closes #78 
Closes #345
Closes #341
Closes #177
Closes #342
Closes #347
Closes #355
Closes #361
Closes #362
Closes #370
Closes #372
Closes #373
Closes #375
Closes #383
Closes #388
Closes #392
Closes #400

### CODE REVIEW

**Comment by JonathonDunford** [comment]
Travis CI failing up to here is an error on its part. The HTTPS of Gifto works fine.

**Comment by JonathonDunford** [comment]
Travis CI erroring out for no reason. The HTTPS of Gifto works fine.

File: tokens/0x63e634330a20150dbb61b15648bc73855d6ccf07.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+decimals: 18
+description: >-
+  Blocklancer is a Distributed Autonomous Job Marketplace (DAJ) operating on the Ethereum Blockchain. Blocklancer
+  is our collaborative vision of an entirely self-regulatory platform for finding clients/projects and ensuring
+  jobs are completed efficiently and fairly. Blocklancer is changing the way freelancing works for both the
+  freelancer and the customer, concentrating on building a trustworthy platform all stakeholders can rely on.
+  Blocklancer does this by minimalizing fees and introducing a decentralized tribunal system to guarantee every
+  dispute case is settled fairly and that no one loses out — we can finally wave goodbye to fraud in freelancing.
+links:
+- Website: http://www.blocklancer.net/
```
Comment (freeatnet) [inline tokens/0x63e634330a20150dbb61b15648bc73855d6ccf07.yaml]: Looks like the website supports HTTPS

File: tokens/0x8b353021189375591723e7384262f45709a3c3dc.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- Linkedin: https://www.linkedin.com/in/chu-duc-minh-1773a229/
+- Linkedin: https://www.linkedin.com/in/chancejdu/
+- Linkedin: https://www.linkedin.com/in/longvuong22
+- Linkedin: https://www.linkedin.com/in/antoine-najjarin-904828b5/
+- Linkedin: https://www.linkedin.com/in/thanhson1085
+- Linkedin: https://www.linkedin.com/in/crypt-xo-androklis-polymenis-6a7137b2/
+- Linkedin: https://www.linkedin.com/in/tung-hoang-719b4657/
+- Linkedin: https://www.linkedin.com/in/minkimd14/
+- Linkedin: https://www.linkedin.com/in/limroger/
+- Linkedin: https://www.linkedin.com/in/hoang-nguyen-a493b158/
```
Comment (freeatnet) [inline tokens/0x8b353021189375591723e7384262f45709a3c3dc.yaml]: Unintentional personal linkedin profiles?

File: tokens/0x95c4be8534d69c248c0623c4c9a7a2a001c17337.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+addr: '0x95c4be8534d69c248c0623c4c9a7a2a001c17337'
+decimals: 18
+description: >-
+  The ultimate Hardware Wallet with the most advanced level of security, designed for easy payments
+  with over 350 cryptocurrency.
+links:
+- Website: http://www.hodler.tech/
```
Comment (freeatnet) [inline tokens/0x95c4be8534d69c248c0623c4c9a7a2a001c17337.yaml]: HTTPS supported. I think.

File: tokens/0xbdcfbf5c4d91abc0bc9709c7286d00063c0e6f22.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- Bitcointalk: https://bitcointalk.org/index.php?topic=2199333.0
+- Blog: https://medium.com/@peerguess
+- CoinMarketCap: https://coinmarketcap.com/currencies/guess/
+- Email: mailto:info@peerguess.com
+- Linkedin: https://www.linkedin.com/in/mertcavdar
+- Linkedin: https://www.linkedin.com/in/jaya-cdn
+- Linkedin: https://www.linkedin.com/in/seyitozgur
+- Linkedin: https://www.linkedin.com/in/mangesh-vyas-596ba234
+- Linkedin: https://www.linkedin.com/in/subhash-patidar-75ba0517
+- Linkedin: https://www.linkedin.com/in/kapil-prajapati-355a2450
```
Comment (freeatnet) [inline tokens/0xbdcfbf5c4d91abc0bc9709c7286d00063c0e6f22.yaml]: Extra linkedin profiles.

File: tokens/0xc5bbae50781be1669306b9e001eff57a2957b09d.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+links:
+- Blog: https://medium.com/@GIFTO
+- CoinMarketCap: https://coinmarketcap.com/currencies/gifto/
+- Email: mailto:info@gifto.io
+- Facebook: https://www.facebook.com/gifto.io/
+- Github: https://github.com/GIFTO-io
+- Reddit: https://www.reddit.com/user/GIFTO-io/
+- Telegram: https://t.me/GIFTOOfficial
+- Twitter: https://twitter.com/GIFTO_io
+- Website: https://gifto.io/
```
Comment (freeatnet) [inline tokens/0xc5bbae50781be1669306b9e001eff57a2957b09d.yaml]: Misconfigured HTTPS causing failure, token team notified. I'd try to pull this token out into a separate PR so we get tests passing on most tokens and a specific reason for one token merged with failing tests.

File: tokens/0xfd107b473ab90e8fbd89872144a3dc92c40fa8c9.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+- Linkedin: https://www.linkedin.com/in/nithin-eapen-630367143/
+- Linkedin: https://www.linkedin.com/in/han-kyul-park-609a0266/
+- Linkedin: https://www.linkedin.com/in/justdeepoo/
+- Linkedin: https://www.linkedin.com/in/karthikn-thangavel/
+- Linkedin: https://www.linkedin.com/in/MyLaLaWorld/
+- Linkedin: https://www.linkedin.com/in/sankalpshangari/
+- Linkedin: https://www.linkedin.com/in/vivek-kumar-tripathi-93190b35/
+- Linkedin: https://www.linkedin.com/in/ds-mohd-amin-bin-abdul-nor-5a8618a8/
+- Linkedin: https://www.linkedin.com/in/sachinshrm/
+- Linkedin: https://www.linkedin.com/in/thomas-mcalister-8a939a128/
```
Comment (freeatnet) [inline tokens/0xfd107b473ab90e8fbd89872144a3dc92c40fa8c9.yaml]: Extra personal linkedin profiles

File: tokens/0x8b353021189375591723e7384262f45709a3c3dc.yaml (hunk 2 above)
Comment (JonathonDunford) [reply tokens/0x8b353021189375591723e7384262f45709a3c3dc.yaml]: Didn't know if we were including team member LIs. Guess not. I'll remove.

File: tokens/0xc5bbae50781be1669306b9e001eff57a2957b09d.yaml (hunk 5 above)
Comment (JonathonDunford) [reply tokens/0xc5bbae50781be1669306b9e001eff57a2957b09d.yaml]: Done: https://github.com/forkdelta/tokenbase/pull/565

**Comment by JonathonDunford** [comment]
@freeatnet Edits made. Travis passing now.

File: tokens/0xd8b980ad351894b8a4cf94856ead06c4586d763f.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  BitSense is a decentralized platform for advertisers, businesses and members. My vision of BitSense is to create
+  a simple ecosystem for everyday users who will be able to earn cryptocurrency/fiat by simply watching daily ads
+  or doing simple tasks. Money they will earn will come from the people advertising their business.
+links:
+- Website: https://www.bitsense.club/
+- Bitcointalk: https://bitcointalk.org/index.php?topic=3013099
+- Twitter: https://twitter.com/bitsenseproject
+- Facebook: https://www.facebook.com/bitsenseproject
+name: BitSense Token
+symbol: BST
```
Comment (freeatnet) [inline tokens/0xd8b980ad351894b8a4cf94856ead06c4586d763f.yaml]: I don't think this is legit.

File: tokens/0xc7a4eb66f982cde71d872398e4d1fc5fc586cb9e.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+addr: '0xc7a4eb66f982cde71d872398e4d1fc5fc586cb9e'
+decimals: 8
+description: >-
+  Rezox Token, is a semi-mineable Token thats based on the Ethereum network. Our goal is to promote low-end
+  computers and make them be able to mine and get our Tokens and take part in the Blockchain revulotion.
```
[the remark this answers is not in the document]
Comment (JonathonDunford) [reply tokens/0xc7a4eb66f982cde71d872398e4d1fc5fc586cb9e.yaml]: This is a comment from the OP here: https://bitcointalk.org/index.php?topic=2912627.0

""Solving math problems" is the simplest explanation of "mining".
We are using Coinhive JS web-miner and reward the miners with Rezox."

[the remark this answers is not in the document]
Comment (mombacoin) [reply tokens/0xc7a4eb66f982cde71d872398e4d1fc5fc586cb9e.yaml]: Everyone who mines sees CoinHive, its "semi-mineable" so its not really mineable but you get the token by mining and if you will look up in Telegram we do not hide this.

**Comment by JonathonDunford** [comment]
Changed something in prep_announce for Discord announcement. Unrelated so forcing the merge.

**Pull request merged**